charts/temporal/templates/server-job.yaml

```diff
@@ -7,6 +7,9 @@ metadata:
     {{- include "temporal.resourceLabels" (list . "database" "") | nindent 4 }}
 spec:
   backoffLimit: {{ $.Values.schema.setup.backoffLimit }}
+  {{- if $.Values.schema.ttlSecondsAfterFinished }}
+  ttlSecondsAfterFinished: {{ $.Values.schema.ttlSecondsAfterFinished }}
```

Comment:
We haven't hit this yet but now I see it being problematic for subsequent schema upgrades. Should we just make this a new default with a sane value like a day? Trying to imagine a scenario where you would want the job to exist after the fact, since the Pod (in our case, due to node rotations) is already gone.

Reply:
I guess that's up for you to decide. We have set it very low (10 seconds) because, especially if some sort of automation is used, updates can happen quite frequently, e.g. after configuration changes.

A more idiomatic approach would be using [chart hooks](https://helm.sh/docs/topics/charts_hooks/), but that doesn't work in this case, because all the deployments wait for the schema to be present but the post-install hook is only executed after deployed resources are ready, which leads to a deadlock situation.

Reply:
There used to be a `pre-install` chart hook on this job which was removed recently in this PR
https://github.com/temporalio/helm-charts/pull/522

I agree with @kosmoz that the chart hook is a better approach, and the `pre-install` hook works perfectly for this.
I would argue for re-introduction of this hook, maybe behind a flag so that people can choose whether they want the job as hook or not.

Reply:
As long as it works, that should be fine by me :-)

Reply:
As detailed in the PR that removes the hooks, they can break anything which uses helm in wait mode, including terraform, pulumi, Argo and somewhat ironically, `helm test`. I will look at this again, but I doubt we'll be able to put hooks back in.

Reply:
Please test #564 as a solution to this issue.